### ISSUE TITLE
Layout for small screens maybe

### ISSUE THREAD

**Issue by erebion** [KRTirtho/spotube#1476]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Pain

The best music app for smartwatches (mostly because there isn't much else :p), but also the most fiddly.

It works when using the compact layout*, but a few layout changes here could make a huge difference.

Login via Spotify is not possible as WearOS does not provide the Webview API for that.

*I also used `wm density 205` via adb shell to reduce screen density a bit from the default 320, which makes it far easier to use the app, as more buttons become visible.

### Suggested solution

- For small screens, like the 320 pixel round screen of the Pixel Watch, order some elements in a different way, maybe remove some buttons to make the app more usable. The app itself works, it's just about the layout. :)
- Maybe add a different login method for Spotify, as webview is not always available. For example on WearOS, maybe it also lacks on other Android versions such as Android TV.

### Self grab

- [ ] I'm ready to work on this issue!

**Comment by erebion** [comment]
Also, once playback has been started, it can comfortably be controlled using the media controls feature of WearOS. :D

**Comment by erebion** [comment]
Here are some screenshots to clarify what I mean:

![image](https://github.com/KRTirtho/spotube/assets/104499031/de795265-f9d0-41f8-b428-7c0f48329084)
![image](https://github.com/KRTirtho/spotube/assets/104499031/c8efd4b9-2f93-4018-a69d-ec85481063a8)
![image](https://github.com/KRTirtho/spotube/assets/104499031/58c71ba1-8021-4eb6-97a3-3b671cc7d0a7)

**Comment by KRTirtho** [comment]
Props on getting it working on a Smart Watch 🎉 . But I wonder why would anyone use a music app on a Smart Watch in the first place? 
At most, the ability to control other devices makes sense but the whole app? Can you even listen to the music? (assuming bluetooth earphones are not connected) 🤔

**Comment by erebion** [comment]
Yes, I can. With and without headphones. My reason is simply going for a walk without much stuff, but having music with me. :)

The built-in speakers might not be the the best for music, but I've used them to listen to a podcast before. Sometimes you don't need high quality.

Also, I imagine people would like music whe running, when a phone would be annoying to have bouncing in a pocket. :)

**Comment by KRTirtho** [comment]
If we have to make an UI, I say, we make a separate watch UI. Cutting off some features and leaving out basic features.
Because, watches are underpowered and has smaller batteries.

Maybe, we should only cover,
- Home page playlists (personal recommendation ones. No generes)
  - The playlists card should shown one at a time.
  - On, vertical scroll it can switch the palylist category
- Have the search icon on home screen
  - Search page, just like home will show one item at time
 - Playlist and albums page will have compacted track details. Just the names, separated by comma in bold text. Kind of how spotify does in the mobile app.
 - Library will only contain liked playlists, and albums. (No, Downloads/Local Tracks. Sorry)
 - Have a Player UI, that comes up when clicking the watch's rotary nob. Or, just have button in the HomePage
   - It'll have all the controls
   - And will have only the Queue
   - We can also put a button to open the lyrics page
- We'll cover all of the settings.

Now, about login, it's the hardest part. We need the WebView for Login flow. Otherwise, user have to manually pass the cookie just like the desktop version.

**Comment by erebion** [comment]
Looks good so far.

> Library will only contain liked playlists, and albums. (No, Downloads/Local Tracks. Sorry)

Not all watches have LTE modems, while going on a walk people might have spotty reception. I consider downloads fo be essential.

> We can also put a button to open the lyrics page

I doubt that would work well and be useful on a tiny watch :)

**Comment by KRTirtho** [comment]
> Not all watches have LTE modems, while going on a walk people might have spotty reception. I consider downloads fo be essential.

Ok, I guess it makes sense include it.

**Comment by sanctimoniousape** [comment]
Gonna piggyback on this because my issue is similar, but on an actual phone (the [Unihertz Jelly Star](https://www.unihertz.com/products/jelly-star), which I came to absolutely ***love*** once I got used to it's diminutive size, and similarly tweaked things like the DPI). The controls on the player screen are cut off and I can't seem to get it to scroll them into view.

![Screenshot_20240818-143330](https://github.com/user-attachments/assets/e6e639ef-8b38-44ae-8f99-da01a4ad3344)
